### PR TITLE
[SPARK-37681][SQL] Support ANSI Aggregate Function: regr_sxy

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR used to support ANSI aggregate Function: `regr_sxy`

The mainstream database supports `regr_sxy` show below:
**Teradata**
https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/MXr3jFyWutZ9J4fhlXv_fQ
**Snowflake**
https://docs.snowflake.com/en/sql-reference/functions/regr_sxy.html
**Oracle**
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/REGR_-Linear-Regression-Functions.html#GUID-A675B68F-2A88-4843-BE2C-FCDE9C65F9A9
**DB2**
https://www.ibm.com/docs/en/db2/11.5?topic=af-regression-functions-regr-avgx-regr-avgy-regr-count
**H2**
http://www.h2database.com/html/functions-aggregate.html#regr_sxy
**Postgresql**
https://www.postgresql.org/docs/8.4/functions-aggregate.html
**Sybase**
https://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.help.sqlanywhere.12.0.0/dbreference/regr-sxy-function.html
**Exasol**
https://docs.exasol.com/sql_references/functions/alphabeticallistfunctions/regr_function.htm

### Why are the changes needed?
`regr_sxy` is very useful.

### Does this PR introduce _any_ user-facing change?
'Yes'. New feature.

### How was this patch tested?
New tests.

### CODE REVIEW

**Comment by beliefer** [comment]
ping @cloud-fan

File: sql/core/src/test/resources/sql-tests/results/postgreSQL/aggregates_part1.sql.out [hunk 1]
```diff
@@ -304,6 +304,14 @@ struct<regr_sxx(b, a):double>
 5099.0
 
 
+-- !query
+SELECT regr_sxy(b, a) FROM aggtest
+-- !query schema
+struct<regr_sxy(b, a):double>
+-- !query output
+2614.5158215500414
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/postgreSQL/aggregates_part1.sql.out:312]: have we verified result with pgsql?

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: sql/core/src/test/resources/sql-tests/results/postgreSQL/aggregates_part1.sql.out (hunk 1 above)
Comment (beliefer) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/aggregates_part1.sql.out:312]: Yes. I verified them.

**Comment by beliefer** [comment]
@cloud-fan Thanks a lot!

**Pull request closed without merging**